openhtf/exe/__init__.py

```diff
@@ -100,22 +74,53 @@ def Stop(self):
         self._exit_stack.close()
     self.Kill()
 
+  def Finalize(self, output_callbacks):
+    """Finalize test execution and output resulting record to callbacks.
+
+    Should only be called once at the conclusion of a test run, and will raise
+    an exception if end_time_millis is already set.
+
+    Raises: TestAlreadyFinalized if end_time_millis already set.
+    """
+    if not self._test_state:
+      raise TestStopError('Test Stopped.')
+
+    if not self._test_state.is_finalized:
+      self._test_state.logger.info('Finishing test with outcome ABORTED.')
+      self._test_state.Finalize(test_record.Outcome.ABORTED)
+
+    _LOG.debug('Test completed for %s, saving to history and outputting.',
```

Comment:
This is the log statement that makes no sense, history isn't affected directly here. If this was moved up into `Test` then it would be logical